hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip551/AtomicBatchInvalidSignaturesTests.java

```diff
@@ -0,0 +1,649 @@
+                    getTxnRecord("failedBatch").logged(),
+                    getTxnRecord(associateTxnId).assertingNothingAboutHashes().logged(),
+
+                    // Verify token and contract exist but no association occurred
+                    getTokenInfo(misc).logged(),
+                    getContractInfo(contract).logged(),
+                    getContractInfo(contract).hasNoTokenRelationship(misc));
+        }
+
+        @HapiTest
+        @DisplayName("Batch with multiple contract associations - mixed admin key scenarios")
+        public Stream<DynamicTest> mixedContractAssociationScenarios() {
+            final var batchOperator = "batchOperator";
+            final var token1 = "token1";
+            final var token2 = "token2";
+            final var contractWithKey = "contractWithKey";
+            final var contractWithoutKey = "contractWithoutKey";
+            final var adminKey = "adminKey";
+            final var associate1TxnId = "associate1TxnId";
+            final var associate2TxnId = "associate2TxnId";
+
+            return hapiTest(
+                    cryptoCreate(batchOperator).balance(ONE_HUNDRED_HBARS),
+                    newKeyNamed(adminKey),
+                    tokenCreate(token1),
+                    tokenCreate(token2),
+                    uploadInitCode("CalldataSize"), // Upload once
+                    contractCreate(contractWithKey).bytecode("CalldataSize").adminKey(adminKey), // Has admin key
+                    contractCreate(contractWithoutKey).bytecode("CalldataSize").omitAdminKey(), // No admin key
+
+                    // Batch mixing valid and invalid associations
+                    atomicBatch(
+                                    // This should work - contract has admin key
+                                    tokenAssociate(contractWithKey, token1)
+                                            .via(associate1TxnId)
+                                            .batchKey(batchOperator)
+                                            .payingWith(batchOperator),
+
+                                    // This should fail - contract has no admin key
+                                    tokenAssociate(contractWithoutKey, token2)
+                                            .via(associate2TxnId)
+                                            .batchKey(batchOperator)
+                                            .payingWith(batchOperator))
+                            .signedByPayerAnd(batchOperator, adminKey)
+                            .via("mixedAssocBatch")
+                            .hasKnownStatus(INNER_TRANSACTION_FAILED),
+                    getTxnRecord("mixedAssocBatch").logged(),
+
+                    // Verify the contracts exist (skip inner record queries that cause RECORD_NOT_FOUND)
+                    getContractInfo(contractWithKey).logged(),
+                    getContractInfo(contractWithoutKey).logged());
```

Comment:
It'd be good to also verify the first inner transaction succeeded. Otherwise, if the first transaction fails, this test isn't testing exactly what we think. A few other tests do this kind of thing too, would be good to verify those as well.